### PR TITLE
Function composition

### PR DESCRIPTION
**Pull request by VrajPatelK**

### CODE REVIEW

File: Function Composition/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/**
+ * @param {Function[]} functions
+ * @return {Function}
+ */
+var compose = function(functions) {
+    return function(x) {
+        if(functions.length === 0) return x;
```
Comment (GauravChopada) [inline Function Composition/index.js:7]: Code will work the same without this condition.

File: Function Composition/index.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        functions.reverse().map((fn)=>{
```
Comment (GauravChopada) [inline Function Composition/index.js:8]: Using reverse makes code look clean here however reverse has complexity of O(n) itself. Here you have scope for optimisation.

**Review by GauravChopada — CHANGES_REQUESTED** [review]

File: Function Composition/index.js (hunk 2 above)
Comment (VrajPatelK) [reply Function Composition/index.js]: **optimization is done !**

**Review by GauravChopada — APPROVED** [review]
Well Done , FYI you can also solve it using reduce method.

**Pull request merged**